File: tests/machine/x/scala/partial_application.scala
object partial_application {
  def add(a: Int, b: Int): Int = {
    return (a).asInstanceOf[Int] + (b).asInstanceOf[Int]
  }
  
  def main(args: Array[String]): Unit = {
    val add5 = (p0: Int) => add(5, p0)
    println((add5(3)))
  }
}

File: tests/machine/x/java/partial_application.java
import java.util.function.*;
public class PartialApplication {
	static int add(int a, int b) {
		return a + b;
	}
	public static void main(String[] args) {
	IntUnaryOperator add5 = b -> add(5, b);
	System.out.println(add5.applyAsInt(3));
	}
}
